lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -66,23 +68,31 @@ class TraceElements extends Gatherer {
   }
 
   /**
+   * This function finds the top (up to 5) elements that contribute to the CLS score of the page.
+   * Each layout shift event has a 'score' which is the amount added to the CLS as a result of the given shift(s).
+   * We calculate the score per element by taking the 'score' of each layout shift event and
+   * distributing it between all the nodes that were shifted, proportianal to the impact region of
+   * each shifted element.
    * @param {Array<LH.TraceEvent>} mainThreadEvents
-   * @return {Array<number>}
+   * @return {Array<TraceElementData>}
    */
-  static getCLSNodeIdsFromMainThreadEvents(mainThreadEvents) {
-    const clsPerNodeMap = new Map();
-    /** @type {Set<number>} */
-    const clsNodeIds = new Set();
+  static getTopLayoutShiftElements(mainThreadEvents) {
```

Comment:
Can you add some unit tests that test this function?

Reply:
That's a great idea. Will do.